models/issue_comment.go

```diff
@@ -55,17 +57,22 @@ const (
 
 // Comment represents a comment in commit and issue page.
 type Comment struct {
-	ID              int64 `xorm:"pk autoincr"`
-	Type            CommentType
-	PosterID        int64 `xorm:"INDEX"`
-	Poster          *User `xorm:"-"`
-	IssueID         int64 `xorm:"INDEX"`
-	LabelID         int64
-	Label           *Label `xorm:"-"`
-	OldMilestoneID  int64
-	MilestoneID     int64
-	OldMilestone    *Milestone `xorm:"-"`
-	Milestone       *Milestone `xorm:"-"`
+	ID             int64 `xorm:"pk autoincr"`
+	Type           CommentType
+	PosterID       int64 `xorm:"INDEX"`
+	Poster         *User `xorm:"-"`
+	IssueID        int64 `xorm:"INDEX"`
+	LabelID        int64
+	Label          *Label `xorm:"-"`
+	OldMilestoneID int64
+	MilestoneID    int64
+	OldMilestone   *Milestone `xorm:"-"`
+	Milestone      *Milestone `xorm:"-"`
+	OldAssigneeID  int64
```

Comment:
It would be nice if we didn't have to have both `OldMilestoneID`/`MilestoneID` and `OldAssigneeID`/`AssigneeID`, since at most one of them will be relevant for any particular issue.

Would it make sense to simply have `OldElementID` and `ElementID` fields? If `Type==CommentTypeAssignees` then `OldElementID` and `ElementID` are assignee IDs, if `Type==CommentTypeMilestone` they are milestone IDs, etc. I see pros and cons, interested to hear your thoughts :smile: